src/python_inspector/resolution.py

```diff
@@ -62,12 +68,49 @@ def get_python_version_from_env_tag(python_version: str):
     return python_version
 
 
+def get_sdist_file(repos, candidate, python_version):
+    """
+    Return the sdist file for a candidate.
```

Comment:
```suggestion
    Fetch and extract the source distribution (sdist) for the ``candidate`` Candidate from the `repos` list of PyPiRepository
    and a required ``python_version`` Python version.
    Return the directory location string where the sdist has been extracted.
    Return None if the sdist was not fetched either because does not exist in any of the ``repos`` or it does not work with 
    the required ``python_version``.
    Raise an Exception if extraction fails. 
```

You shuld IMHO return the   
```
os.path.join(utils_pypi.CACHE_THIRDPARTY_DIR,
                "extracted_sdists",
                sdist_file,
                sdist_file)
```